### PR TITLE
Feature: Download for V2 datasets

### PR DESCRIPTION
**Pull request by cwognum**
## Changelogs

- Added support for downloading V2 Datasets from the Hub.
- Given that `authlib >1.3.0` is [not yet available for Conda](https://github.com/conda-forge/authlib-feedstock), we can't assume a [recently merged QoL improvement](https://github.com/lepture/authlib/commit/16fa567110b4bb4094f3ab3a452bafb9945847fb) is available and needed some code to handle the case that `token` is None in `ensure_active_token()`.
- Excluded some fields from the `repr_dict` to prevent excessive compute when visualizing the dataset.

---

_Checklist:_

- [X] _Was this PR discussed in an issue? It is recommended to first discuss a new feature into a GitHub issue before opening a PR._
- [ ] ~_Add tests to cover the fixed bug(s) or the newly introduced feature(s) (if appropriate)._~
- [X] _Update the API documentation if a new function is added, or an existing one is deleted._
- [X] _Write concise and explanatory changelogs above._
- [X] _If possible, assign one of the following labels to the PR: `feature`, `fix`, `chore`, `documentation` or `test` (or ask a maintainer to do it for you)._

---

The `client.get_dataset()` tries loading a V1 dataset first, and then the V2 if a 404 is raised. This adds some overhead, but given the deadline I would like us to optimize this later. 

### A word on testing

We don't have a great way setup yet to test these changes, so I manually created three datasets on the `cas` Neon branch: 

1. A vanilla V1 dataset
2. A V1 dataset with pointer columns
3. A V2 dataset

All three of these datasets are public, so as long as you use my Neon branch you should be able to replicate the tests I've done using this code:
```python
with PolarisHubClient(settings=settings) as client:
    # V2 dataset
    dataset = client.get_dataset(owner="casw3", name="BELKA-v5")
    print(dataset[123456])

    # V1 dataset with pointers
    dataset = client.get_dataset(owner="casw3", name="v1-with-pointers-2")
    print(dataset[0])

    # Vanilla V1 dataset
    dataset = client.get_dataset(owner="casw3", name="biogen-adme-v1")
    print(dataset[123])
```

This also works:
```py
for name in ["BELKA-v5", "v1-with-pointers-2", "biogen-adme-v1"]:
    po.load_dataset(f"casw3/{name}")
```

### CODE REVIEW

File: polaris/dataset/_base.py [hunk 1]
```diff
@@ -175,31 +174,48 @@ def zarr_root(self) -> zarr.Group | None:
...
+                    f"using {self.__class__.__name__}.cache() for more performant data access."
                 )
                 self._warn_about_remote_zarr = False
 
         try:
             if saved_on_hub:
-                with PolarisHubClient() as client:
-                    with StorageSession(client, "read", self.urn) as storage:
-                        self._zarr_root = zarr.open_consolidated(storage.extension_store)
+                self._zarr_root = self.load_zarr_root_from_hub()
```
Comment (jstlaurent) [inline polaris/dataset/_base.py:188]: **Comment**
This is nice, much cleaner. Getting closer to a repository pattern.

File: polaris/hub/client.py [hunk 2]
```diff
@@ -802,7 +832,7 @@ def get_competition(
 
         # TODO (jstlaurent): response["dataset"]["artifactId"] is the owner/name unique identifier,
         #  but we'd need to change the signature of get_dataset to use it
-        response["dataset"] = self._get_dataset(
+        response["dataset"] = self._get_v1_dataset(
             response["dataset"]["owner"]["slug"],
             response["dataset"]["name"],
             ArtifactSubtype.COMPETITION,
```
Comment (mercuryseries) [inline polaris/hub/client.py:838]:
```suggestion
            ArtifactSubtype.COMPETITION.value,
```

File: polaris/hub/client.py [hunk 3]
```diff
@@ -324,39 +339,54 @@ def _get_dataset(
...
+            if zarr_root_path is not None:
+                # For V1 datasets, the Zarr Root is optional.
+                # It should be None if the dataset does not use pointer columns
+                zarr_root_path = str(zarr_root_path)
 
-            if dataset.should_verify_checksum(verify_checksum):
-                dataset.verify_checksum(md5sum)
-            else:
-                dataset.md5sum = md5sum
+        if artifact_type == ArtifactSubtype.COMPETITION:
```
Comment (mercuryseries) [inline polaris/hub/client.py:362]: Not directly related to your PR, but I noticed we’re using `ArtifactSubtype.STANDARD.value` in other places, so:

```suggestion
        if artifact_type == ArtifactSubtype.COMPETITION.value:
```

**Review by mercuryseries — APPROVED** [review]
Thanks @cwognum! Looks great!

File: polaris/hub/client.py [hunk 4]
```diff
@@ -304,9 +310,18 @@ def get_dataset(
...
             A `Dataset` instance, if it exists.
         """
-        return self._get_dataset(owner, name, ArtifactSubtype.STANDARD.value, verify_checksum)
+        with ProgressIndicator(
+            start_msg="Fetching dataset...",
+            success_msg="Fetched dataset.",
+            error_msg="Failed to fetch dataset.",
+        ):
+            try:
+                return self._get_v1_dataset(owner, name, ArtifactSubtype.STANDARD.value, verify_checksum)
```
Comment (mercuryseries) [inline polaris/hub/client.py:319]: The `artifact_type` param is typed as `ArtifactSubtype`, but we pass a string in most places.

```suggestion
                return self._get_v1_dataset(owner, name, ArtifactSubtype.STANDARD, verify_checksum)
```

**Pull request merged**